dev/integration_tests/windows_startup_test/windows/runner/flutter_window.cpp

```diff
@@ -50,11 +64,44 @@ bool FlutterWindow::OnCreate() {
       &flutter::StandardMethodCodec::GetInstance());
 
   channel.SetMethodCallHandler(
-    [](const flutter::MethodCall<>& call,
+    [&](const flutter::MethodCall<>& call,
        std::unique_ptr<flutter::MethodResult<>> result) {
-      std::scoped_lock lock(visible_mutex);
-      if (call.method_name() == "isWindowVisible") {
+       std::string method = call.method_name();
+
+      if (method == "isWindowVisible") {
+        std::scoped_lock lock(visible_mutex);
         result->Success(visible);
+      } else if (method == "isAppDarkModeEnabled") {
+        BOOL enabled;
+        HRESULT hr = DwmGetWindowAttribute(GetHandle(),
+                                           DWMWA_USE_IMMERSIVE_DARK_MODE,
+                                           &enabled, sizeof(enabled));
+        if (SUCCEEDED(hr)) {
+          result->Success((bool)enabled);
+        } else if (hr == E_INVALIDARG) {
+          // Fallback if the operating system doesn't support dark mode.
+          result->Success(false);
+        } else {
+          result->Error("error", "Received result handle " + hr);
+        }
+      } else if (method == "isSystemDarkModeEnabled") {
+        DWORD data;
+        DWORD data_size = sizeof(data);
+        LONG status = RegGetValue(HKEY_CURRENT_USER,
+                                  kGetPreferredBrightnessRegKey,
+                                  kGetPreferredBrightnessRegValue,
+                                  RRF_RT_REG_DWORD, nullptr, &data, &data_size);
+
+        if (status == ERROR_SUCCESS) {
+          // Preferred brightness is 0 if dark mode is enabled,
+          // otherwise non-zero.
+          result->Success(data == 0);
+        } else if (status == ERROR_FILE_NOT_FOUND) {
+          // Fallback if the operating system doesn't support dark mode.
+          result->Success(false);
```

Comment:
This is part of the fix. See: https://github.com/flutter/flutter/commit/662a714fb369881536209aaf332bf4906c570a35